src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -426,20 +433,72 @@ internal async Task WriteAsGnuAsync(Stream archiveStream, Memory<byte> buffer, C
             }
         }
 
+        private static MemoryStream GetLongMetadataStream(string text)
+        {
+            MemoryStream data = new MemoryStream();
+            data.Write(Encoding.UTF8.GetBytes(text));
+            data.WriteByte(0); // Add a null terminator at the end of the string, _size will be calculated later
+            data.Position = 0;
+            return data;
+        }
+
+        private static async ValueTask<MemoryStream> GetLongMetadataStreamAsync(string text, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            MemoryStream data = new MemoryStream();
+            await data.WriteAsync(Encoding.UTF8.GetBytes(text), cancellationToken).ConfigureAwait(false);
```

Comment:
I don't think you need to add the async versions, you're creating the MemoryStream so there won't be a derived type and it will simply call Write from WriteAsync internally.

Reply:
That simplifies the code a lot.

Reply:
Fixed in main, now backporting.